src/back/space/src/repo/table_space.cpp

```diff
@@ -0,0 +1,278 @@
+#include <userver/storages/postgres/component.hpp>
+
+#include <boost/uuid/uuid_io.hpp>
+
+namespace svetit::space::table {
+
+namespace pg = storages::postgres;
+
+Space::Space(storages::postgres::ClusterPtr pg)
+	: _pg{std::move(pg)}
+{
+	constexpr auto kCreateTable = R"~(
+CREATE TABLE IF NOT EXISTS space (
+	id UUID PRIMARY KEY,
+	name TEXT NOT NULL,
+	key TEXT NOT NULL,
+	requestsAllowed BOOLEAN NOT NULL,
+	createdAt BIGINT NOT NULL
+);
+)~";
+
+	using storages::postgres::ClusterHostType;
+	_pg->Execute(ClusterHostType::kMaster, kCreateTable);
+
+	//InsertDataForMocks();
+}
+
+const storages::postgres::Query kInsertSpace{
+	"INSERT INTO space (id, name, key, requestsAllowed, createdAt) "
+	"VALUES ($1, $2, $3, $4, $5) ",
+	storages::postgres::Query::Name{"insert_space"},
+};
+
+void Space::Insert(
+	const boost::uuids::uuid& uuid,
+	const std::string& name,
+	const std::string& key,
+	const bool requestsAllowed,
+	const int64_t createdAt)
+{
+	_pg->Execute(storages::postgres::ClusterHostType::kMaster, kInsertSpace, uuid, name, key, requestsAllowed, createdAt);
+}
+
+const storages::postgres::Query kSelectSpace{
+	"SELECT id, name, key, requestsAllowed, createdAt "
+	"FROM space OFFSET $1 LIMIT $2",
+	storages::postgres::Query::Name{"select_space"},
+};
+
+std::vector<model::Space> Space::Select(const int offset, const int limit)
+{
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectSpace, offset, limit);
+
+	if (res.IsEmpty())
+	{
+		return {};
+	}
+
+	return res.AsContainer<std::vector<model::Space>>(pg::kRowTag);
+}
+
+const storages::postgres::Query kSelectSpaceAvailable{
+	R"~(
+		SELECT id, name, key, requestsAllowed, createdAt
+		FROM space
+		WHERE requestsAllowed = true
+		AND id NOT IN (
+			SELECT spaceId FROM space_user WHERE userId = $1
+		) OFFSET $2 LIMIT $3
+	)~",
+	storages::postgres::Query::Name{"select_space_available"},
+};
+
+std::vector<model::Space> Space::SelectAvailable(const std::string& userId, const int offset, const int limit)
+{
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectSpaceAvailable, userId, offset, limit);
+
+	if (res.IsEmpty())
+	{
+		return {};
+	}
+
+	return res.AsContainer<std::vector<model::Space>>(pg::kRowTag);
+}
+
+const storages::postgres::Query kSelectByUserId{
+	R"~(
+		SELECT id, name, key, requestsAllowed, createdAt
+		FROM space
+		WHERE id IN (
+			SELECT spaceId FROM space_user WHERE userId = $1
+		) OFFSET $2 LIMIT $3
+	)~",
+	storages::postgres::Query::Name{"select_space_by_user_id"},
+};
+
+std::vector<model::Space> Space::SelectByUserId(const std::string& userId, const int offset, const int limit)
+{
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectByUserId, userId, offset, limit);
+
+	if (res.IsEmpty())
+	{
+		return {};
+	}
+
+	return res.AsContainer<std::vector<model::Space>>(pg::kRowTag);
+}
+
+const storages::postgres::Query kCountSpace{
+	"SELECT count(id) FROM space",
```

Comment:
Давай использовать звёздочку вместо имени pk
```suggestion
	"SELECT COUNT(*) FROM space",
```

Reply:
Готово.